kbase-extension/static/kbase/js/widgets/appWidgets2/validation.js

```diff
@@ -302,41 +326,63 @@ define([
 
         function validateFloat(value, min, max) {
             if (isNaN(value)) {
-                return 'value must be numeric';
+                return {
+                    messageId: Constants.MESSAGE_IDS.NAN,
+                    errorMessage: 'value must be numeric',
+                };
             }
             if (!isFinite(value)) {
-                return 'value must be a finite float';
+                return {
+                    messageId: Constants.MESSAGE_IDS.INFINITE,
+                    errorMessage: 'value must be finite',
+                };
             }
-            if (max && max < value) {
-                return 'the maximum value for this parameter is ' + max;
+            if (max && value > max) {
+                return {
+                    messageId: Constants.MESSAGE_IDS.VALUE_OVER_MAX,
+                    errorMessage: `the maximum value for this parameter is ${max}`,
+                };
             }
-            if (min && min > value) {
-                return 'the minimum value for this parameter is ' + min;
+            if (min && value < min) {
+                return {
+                    messageId: Constants.MESSAGE_IDS.VALUE_UNDER_MIN,
+                    errorMessage: `the minimum value for this parameter is ${min}`,
+                };
             }
         }
 
-        function validateFloatString(value, constraints) {
-            let normalizedValue, parsedValue, errorMessage, diagnosis;
+        function validateFloatString(value, constraints, options = {}) {
+            let plainValue, parsedValue, errorMessage, messageId, diagnosis;
```

Comment:
Changed to `plainValue`, keep in light with `validateIntString`.

TODO: merge `validateFloatString` and `validateIntString` -> `validateNumberString` with an extra option.